smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/AddSdkStreamMixinDependency.java

```diff
@@ -0,0 +1,121 @@
+public final class AddSdkStreamMixinDependency implements TypeScriptIntegration {
+
+    @Override
+    public void addConfigInterfaceFields(
+        TypeScriptSettings settings,
+        Model model,
+        SymbolProvider symbolProvider,
+        TypeScriptWriter writer
+    ) {
+        if (!hasStreamingBlobDeser(settings, model)) {
+            return;
+        }
+
+        writer.addImport("SdkStreamMixinInjector", "__SdkStreamMixinInjector",
+                TypeScriptDependency.AWS_SDK_TYPES.packageName);
+        writer.writeDocs("The internal function that inject utilities to runtime-specific stream to help users"
+                + " consume the data\n@internal");
+        writer.write("sdkStreamMixin?: __SdkStreamMixinInjector;\n");
+    }
+
+    @Override
+    public Map<String, Consumer<TypeScriptWriter>> getRuntimeConfigWriters(
+        TypeScriptSettings settings,
+        Model model,
+        SymbolProvider symbolProvider,
+        LanguageTarget target
+    ) {
+        if (!hasStreamingBlobDeser(settings, model)) {
+            return Collections.emptyMap();
+        }
+        switch (target) {
+            case NODE:
+                return MapUtils.of("sdkStreamMixin", writer -> {
+                    writer.addDependency(TypeScriptDependency.UTIL_STREAM_NODE);
+                    writer.addImport("sdkStreamMixin", "sdkStreamMixin",
+                            TypeScriptDependency.UTIL_STREAM_NODE.packageName);
+                    writer.write("sdkStreamMixin");
+                });
+            case BROWSER:
+                return MapUtils.of("sdkStreamMixin", writer -> {
+                    writer.addDependency(TypeScriptDependency.UTIL_STREAM_BROWSER);
+                    writer.addImport("sdkStreamMixin", "sdkStreamMixin",
+                            TypeScriptDependency.UTIL_STREAM_BROWSER.packageName);
+                    writer.write("sdkStreamMixin");
+                });
+            default:
+                return Collections.emptyMap();
+        }
+    }
+
+    private static boolean hasStreamingBlobDeser(TypeScriptSettings settings, Model model) {
+        ServiceShape serviceShape = settings.getService(model);
+        TopDownIndex topDownIndex = TopDownIndex.of(model);
+        Set<OperationShape> operations = topDownIndex.getContainedOperations(serviceShape);
+        for (OperationShape operation : operations) {
+            if (hasStreamingBlobDeser(settings, model, operation)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static boolean hasStreamingBlobDeser(TypeScriptSettings settings, Model model, OperationShape operation) {
+        StructureShape ioShapeToDeser;
+        if (settings.generateServerSdk()) {
+            ioShapeToDeser = model.expectShape(operation.getInputShape()).asStructureShape().get();
+        } else {
+            ioShapeToDeser = model.expectShape(operation.getOutputShape()).asStructureShape().get();
+        }
```

Comment:
nit: use ternary operator
```suggestion
        StructureShape ioShapeToDeser = (settings.generateServerSdk())
          ? model.expectShape(operation.getInputShape()).asStructureShape().get()
          : model.expectShape(operation.getOutputShape()).asStructureShape().get();
```